NCS.DSS.EmploymentProgression/PatchEmploymentProgression/Service/EmploymentProgressionPatchTriggerService.cs

```diff
@@ -0,0 +1,151 @@
+
+                if (employmentProgressionPatch.DateOfLastEmployment.HasValue)
+                    _jsonHelper.UpdatePropertyValue(employmentProgressionAsJsonObject["DateOfLastEmployment"], employmentProgressionPatch.DateOfLastEmployment);
+
+                if (employmentProgressionPatch.LengthOfUnemployment.HasValue)
+                    _jsonHelper.UpdatePropertyValue(employmentProgressionAsJsonObject["LengthOfUnemployment"], employmentProgressionPatch.LengthOfUnemployment);
+
+                if (employmentProgressionPatch.LastModifiedDate.HasValue)
+                    _jsonHelper.UpdatePropertyValue(employmentProgressionAsJsonObject["LastModifiedDate"], employmentProgressionPatch.LastModifiedDate);
+
+                if (!string.IsNullOrEmpty(employmentProgressionPatch.LastModifiedTouchpointId))
+                    _jsonHelper.UpdatePropertyValue(employmentProgressionAsJsonObject["LastModifiedTouchpointId"], employmentProgressionPatch.LastModifiedTouchpointId);
+
+                if (employmentProgressionPatch.EmploymentHours.HasValue)
+                    _jsonHelper.UpdatePropertyValue(employmentProgressionAsJsonObject["EmploymentHours"], employmentProgressionPatch.EmploymentHours);
+
+                return employmentProgressionAsJsonObject.ToString();
+            }
+            catch (JsonReaderException)
+            {
+                return null;
+            }
+        }
+
+        public async Task<Models.EmploymentProgression> UpdateCosmosAsync(string employmentProgressionAsJson, Guid employmentProgressionId)
+        {
+            if (string.IsNullOrEmpty(employmentProgressionAsJson))
+            {
+                return null;
+            }
+
+            var response = await _documentDbProvider.UpdateEmploymentProgressionAsync(employmentProgressionAsJson, employmentProgressionId);
+            var responseStatusCode = response?.StatusCode;
+
+            return responseStatusCode == HttpStatusCode.OK ? (dynamic)response.Resource : null;
+        }
+
+        public async Task SendToServiceBusQueueAsync(Models.EmploymentProgression employmentProgression, string reqUrl)
+        {
+            await _serviceBusClient.SendPostMessageAsync(employmentProgression, reqUrl);
+        }
+
+        public bool DoesEmploymentProgressionExistForCustomer(Guid customerId)
+        {
+            return _documentDbProvider.DoesEmploymentProgressionExistForCustomer(customerId);
+        }
+
+        public async Task<bool> DoesCustomerExist(Guid customerId)
+        {
+            return await _documentDbProvider.DoesCustomerResourceExist(customerId);
+        }
+
+        public void SetIds(EmploymentProgressionPatch employmentProgressionPatch, Guid employmentProgressionGuid, string touchpointId)
+        {
+            employmentProgressionPatch.LastModifiedTouchpointId = touchpointId;
+            employmentProgressionPatch.EmploymentProgressionId = employmentProgressionGuid;
```

Comment:
Move to the model for consistency  set ids and long and lat

Reply:
Agreed to keep functionality in Service.